File: dynabuffers-python/dynabuffers/ast/annotation/NotBlank.py
from dynabuffers.api.IAnnotation import IAnnotation
from dynabuffers.ast.structural import Value


class NotBlank(IAnnotation):

    def __init__(self, args:[Value]):
        self.args = args
        if len(args) != 0:
            raise ValueError("too much annotation values")

    def validate(self, fieldName, value):
        if len(str(value).strip()) == 0:
            raise ValueError("(" + str(fieldName) + ":" + str(value) + ") is blank")

File: dynabuffers-java/src/main/kotlin/dynabuffers/ast/annotation/NotBlank.kt
package dynabuffers.ast.annotation

import dynabuffers.api.IAnnotation
import dynabuffers.ast.structural.Value
import dynabuffers.exception.DynabuffersException

class NotBlank(val args:List<Value>) : IAnnotation {

    init {
        require(args.isEmpty())
    }

    override fun validate(fieldName:String, obj:Any?) {
        if (obj.toString().isBlank()) {
            throw DynabuffersException("($fieldName:'$obj') value is blank")
        }
    }
}
